faunadb-scala/src/main/scala/faunadb/query/Language.scala

```diff
@@ -36,189 +43,167 @@ object Language {
     case object Delete extends Action("delete")
   }
 
-  implicit def stringToObjectPath(str: String) = ObjectPath(str)
-  implicit def intToArrayPath(i: Int) = ArrayPath(i)
-  implicit def stringToValue(unwrapped: String) = StringV(unwrapped)
-  implicit def intToValue(unwrapped: Int) = NumberV(unwrapped)
-  implicit def longToValue(unwrapped: Long) = NumberV(unwrapped)
-  implicit def boolToValue(unwrapped: Boolean) = BooleanV(unwrapped)
-  implicit def arrayToValue(unwrapped: Array[Value]) = ArrayV(unwrapped)
-  implicit def mapToValue(unwrapped: collection.Map[String, Value]) = ObjectV(unwrapped)
-  implicit def doubleToValue(unwrapped: Double) = DoubleV(unwrapped)
-  implicit def pairToValuePair[T](p: (String, T))(implicit convert: T => Value) = {
-    (p._1, convert(p._2))
-  }
+  /**
+    * Helper for path syntax
+    */
+  protected class Path(val segments: Value*) { def /(seg: Value) = new Path(segments :+ seg: _*) }
+  implicit def strToPath(str: String) = new Path(StringV(str))
+  implicit def intToPath(int: Int) = new Path(NumberV(int))
+
+  case class LambdaArg(val params: Value, val expr: Value)
+
+  implicit def arrayLambdaArg(t: (ArrayV, Value)) =
+    t match { case (as, expr) => LambdaArg(as, expr) }
+
+  implicit def arity1LambdaArg(t: (String, Value)) =
+    t match { case (a, expr) => LambdaArg(a, expr) }
+
+  // FIXME: clean these up with a macro to support all tuple arities.
```

Comment:
think its worth just doing this now? i doubt people will have a >5ary method so i dont really care